controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -99,11 +137,481 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Contex
 }
 
 func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+	if !r.storageConsumer.Spec.Enable {
+		r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateDisabled)
+		return reconcile.Result{}, nil
+	}
+
+	r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateConfiguring)
+
+	for _, cephResourceSpec := range r.storageClassClaim.Status.CephResources {
+		r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+	}
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+		if !contains(r.storageClassClaim.GetFinalizers(), storageClassClaimFinalizer) {
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+			r.storageClassClaim.ObjectMeta.Finalizers = append(r.storageClassClaim.ObjectMeta.Finalizers, storageClassClaimFinalizer)
+			if err := r.Client.Update(ctx, r.storageClassClaim); err != nil {
+				r.log.Error(err, "Failed to update StorageClassClaim with finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+				return reconcile.Result{}, err
+			}
+		}
+		if r.storageClassClaim.Spec.Type == "cephBlockPool" {
```

Comment:
https://github.com/red-hat-storage/ocs-operator/blob/bf0eca86ab49d55958de356ca5e6a1fa9f44b02e/api/v1alpha1/storageclassclaim_types.go#L27-L28
```suggestion
		if r.storageClassClaim.Spec.Type == "blockpool" {
```

Reply:
ack